miden-crypto/src/encryption/aes_gcm/mod.rs

```diff
@@ -0,0 +1,263 @@
+use alloc::vec::Vec;
+
+use aes_gcm::{
+    Aes256Gcm, Error,
+    aead::{
+        Aead, AeadCore, KeyInit,
+        rand_core::{CryptoRng, RngCore},
+    },
+};
+use sha3::digest::consts::U12;
+use winter_utils::{ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable};
+
+use crate::{
+    Felt,
+    encryption::{BINARY_CHUNK_SIZE, rpo::EncryptionError},
+};
+
+#[cfg(test)]
+mod test;
+
+/// A 96-bit nonce
+#[derive(Debug, PartialEq, Eq)]
+pub struct Nonce {
+    inner: aes_gcm::Nonce<U12>,
+}
+
+impl Nonce {
+    /// Creates a new random nonce using the provided random number generator
+    pub fn with_rng<R: RngCore + CryptoRng>(rng: &mut R) -> Self {
+        Nonce { inner: Aes256Gcm::generate_nonce(rng) }
+    }
+
+    /// Creates a new nonce from the provided array of bytes
+    pub fn from_slice(bytes: &[u8; 12]) -> Self {
+        Nonce { inner: (*bytes).into() }
+    }
+}
+
+/// A 256-bit secret key represented
+#[derive(Debug, PartialEq, Eq)]
+pub struct SecretKey([u8; 32]);
```

Comment:
If we need the secret key to be a word, then it might make sense to make the inner type a Word to begin with